### PR TITLE
Prevent chain service from subscribing to networks twice.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
This should also help decrease our json-rpc usage.

### To Test
- [x] Drop a setInterval inside of chain service that logs `this.subscribedNetworks `
- [x] Ensure that the `this.subscribedNetworks ` array does not have any duplicates.

### CODE REVIEW

File: background/services/chain/tests/index.integration.test.ts [hunk 1]
```diff
@@ -32,6 +32,24 @@ describe("ChainService", () => {
...
+    it("should not add duplicate networks on startup", async () => {
+      // Startup is simulated in the `beforeEach`
+      expect(
+        chainService.supportedNetworks.filter(
+          (network) => network.chainID === ETHEREUM.chainID
+        )
+      ).toHaveLength(1)
+
+      expect(
+        chainService.supportedNetworks.filter(
```
Comment (hyphenized) [inline background/services/chain/tests/index.integration.test.ts:46]: should this be `subscribedNetworks`?

**Review by hyphenized — COMMENTED** [review]
Looks good! I can confirm subscriptions are only happening once per network now 👀 👀

**Review by jagodarybacka — APPROVED** [review]
LGTM 👾

File: background/services/chain/index.ts [hunk 2]
```diff
@@ -268,13 +268,6 @@ export default class ChainService extends BaseService<Events> {
     const activeNetworks = await this.getActiveNetworks()
 
     // get the latest blocks and subscribe for all active networks
-    // TODO revisit whether we actually want to subscribe to new heads
-    // if a user isn't tracking a relevant addressOnNetwork
-    activeNetworks.forEach(async (network) => {
```
Comment (greg-nagy) [inline background/services/chain/index.ts:273]: ❤️

**Pull request merged**